DigitalLearningSolutions.Web/ServiceFilter/ValidateAllowedDlsSubApplication.cs

```diff
@@ -0,0 +1,182 @@
+﻿namespace DigitalLearningSolutions.Web.ServiceFilter
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Claims;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Web.Controllers;
+    using DigitalLearningSolutions.Web.Controllers.LearningSolutions;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+    using Microsoft.AspNetCore.Mvc.ModelBinding;
+    using Microsoft.FeatureManagement;
+
+    public class ValidateAllowedDlsSubApplication : IActionFilter
+    {
+        private readonly IFeatureManager featureManager;
+        private readonly IEnumerable<DlsSubApplication> validApplications;
+
+        public ValidateAllowedDlsSubApplication(
+            IFeatureManager featureManager,
+            string[]? validApplicationNames = null
+        )
+        {
+            validApplications = validApplicationNames == null
+                ? new DlsSubApplication[] { }

```

Comment:
Think it would be slightly nicer to use `Enumeration.GetAll<DlsSubApplication>()` in the default case. Then ValidApplications has a more consistent meaning, and you don't need the `.Any()` in `ApplicationIsInaccessibleByPage()`